source/Phrase.cpp

```diff
@@ -34,7 +54,22 @@ void Phrase::Load(const DataNode &node)
 		if(child.Token(0) == "word")
 		{
 			for(const DataNode &grand : child)
-				part.words.push_back(grand.Token(0));
+			{
+				const string &word = grand.Token(0);
+				bool error = false;
+				Format::Replace(word, [&](const string &s) -> string
```

Comment:
Can we improve performance here by also calculating if `word` needs runtime replacement? Then we need not always call phrase expansion unless `word` actually needs it.

Reply:
I'll change the class Part to this in order to avoid runtime replacement:
```
	class Part {
	public:
		std::vector<std::vector<std::pair<std::string, const Phrase *>>> words;
		std::vector<std::function<std::string(const std::string&)>> replaceRules;
	};
```
Phrase::Load() converts a word including phrase expansions to a vector *words* in advance.
Format::Replace I created will specialize in parsing word and be moved into Phrase.cpp.